src/List/ItemList.js

```diff
@@ -12,11 +12,34 @@ const Items = props => {
       db.collection(token)
         .get()
         .then(querySnapshot => {
-          const data = querySnapshot.docs.map(doc => doc.data());
+          const data = querySnapshot.docs.map(doc => {
+            return doc.data();
+          });
           setdbItems(data);
         });
     }
-  }, [token, setdbItems, props]);
+  }, [token, setdbItems, dbItems, props]);
+
+  //This will update an item to include a purchase date
+  const handleChange = e => {
+    let datePurchased = new Date();
+    let db = firebase.fb.firestore();
+    db.collection(token)
+      .doc(e.target.value)
+      .update({ datePurchased });
+    console.log(dbItems);
+  };
+
+  const hours24 = 86400; //24 hours in seconds
+  //A checked box will remain true for 24 hours
+  const is24Hours = item => {
+    let newDay = new Date();
+    if (item.datePurchased) {
+      return newDay.getTime() / 1000 - item.datePurchased.seconds < hours24;
+    } else {
```

Comment:
One interesting thing to consider about `else` here: If you wrote this function like:

```js
if (item.datePurchased) {
  return newDay.getTime() / 1000 - item.datePurchased.seconds < hours24;
}

return false;
```

It would be functionally the same, but with less to read. The `else` is nice for when your `if` statement doesn't return, but because it _does_ in fact return, it's not needed 😄

Reply:
Oh, right. And that’s how we had it but I was messing around and didn’t put it back when I was done lol